pkg/internal/testing/controlplane/etcd.go

```diff
@@ -150,7 +167,7 @@ func (e *Etcd) Stop() error {
 
 func (e *Etcd) defaultArgs() map[string][]string {
 	args := map[string][]string{
-		"listen-peer-urls": {"http://localhost:0"},
+		"listen-peer-urls": {e.listenPeerURL.String()},
```

Comment:
What difference does this make? Shouldn't etcd get a free port from the kernel (and then our allocation not consider that one)?

Reply:
If there are multiple parallel packages running at the same time, process A could internally allocate a port (on file) which is then used for either etcd listen or api-server, while process B (etcd) is already listening on it. 

There is always a race condition between the code that allocates the port and the actual program startup, because the port gets closed right after we have a file lock, another program not knowing about the file allocation can snatch it.

Reply:
Makes sense, thanks for the explanation.

/lgtm
/approve